src/main/java/org/swisspush/redisques/action/GetQueueItemsAction.java

```diff
@@ -27,20 +27,31 @@ public GetQueueItemsAction(Vertx vertx, RedisProvider redisProvider, String addr
 
     @Override
     public void execute(Message<JsonObject> event) {
-        String queueName = event.body().getJsonObject(PAYLOAD).getString(QUEUENAME);
-        String keyListRange = queuesPrefix + queueName;
-        int maxQueueItemCountIndex = getMaxQueueItemCountIndex(event.body().getJsonObject(PAYLOAD).getString(LIMIT));
+        final String queueName = event.body().getJsonObject(PAYLOAD).getString(QUEUENAME);
+        final String queueKey = queuesPrefix + queueName;
+        final int maxQueueItemCountIndex = getMaxQueueItemCountIndex(event.body().getJsonObject(PAYLOAD).getString(LIMIT));
 
-        redisProvider.redis().onSuccess(redisAPI -> redisAPI.llen(keyListRange).onSuccess(countReply -> {
-            Long queueItemCount = countReply.toLong();
-            if(queueItemCount != null) {
-                redisAPI.lrange(keyListRange, "0", String.valueOf(maxQueueItemCountIndex),
-                        new GetQueueItemsHandler(event, queueItemCount));
-            } else {
-                event.reply(exceptionFactory.newReplyException(
-                    "Operation getQueueItems failed to extract queueItemCount", null));
-            }
-        }).onFailure(throwable -> handleFail(event, "Operation getQueueItems failed", throwable)))
+        redisProvider.redis().onSuccess(redisAPI -> redisAPI.llen(queueKey).onSuccess(countReply -> {
+                    Long queueItemCount = countReply.toLong();
+                    if (queueItemCount != null) {
+                        final QueueConfiguration queueConfiguration = findQueueConfiguration(queueName);
+
+                        if (queueConfiguration != null && queueConfiguration.getMaxQueueEntries() > 0) {
+                            final int maxQueueEntries = queueConfiguration.getMaxQueueEntries();
+                            long startRange = Math.max(0, queueItemCount - maxQueueEntries);
+                            long endRange = queueItemCount;
+                            long endIndex = Math.min(startRange + maxQueueItemCountIndex, endRange);
+                            redisAPI.lrange(queueKey, String.valueOf(startRange), String.valueOf(endIndex),
```

Comment:
If  a queue doesn't have a consumer it will not be trimmed, so need lie on those numbers

Reply:
I am not sure if I like this approach. The API will not return the actual state of the queue anymore when you "hide" queue items that are actually present. Reading the queue should return the truth, otherwise it makes debugging and analysis really hard.